### DISCUSSION TITLE
Components from Uno Gallery does not work with net7 Xaml Parsing error

### DISCUSSION THREAD

**Discussion by michalss** [unoplatform/uno#10506] [Q&A]
### Current behavior

Hi i did try a few components from your sample uno gallery page. And i have to say lots of them does not work. Simple create a new project and add this code to MainPage.xaml file. Using latest template and plugin to VS 2022.. Did try WinUI, Skia, WPF, none of them work. No any error before i run it..

![image](https://user-images.githubusercontent.com/1000729/202909507-b7333750-0601-42af-b566-73ebb8feefee.png)

```csharp
        <GridView ItemsSource="123456781234567812345678"
          HorizontalAlignment="Center">
            <GridView.ItemContainerStyle>
                <Style TargetType="GridViewItem">
                    <Setter Property="VerticalContentAlignment"
                 Value="Stretch" />
                    <Setter Property="HorizontalContentAlignment"
                 Value="Stretch" />
                </Style>
            </GridView.ItemContainerStyle>
            <GridView.ItemTemplate>
                <DataTemplate>
                    <!-- Background -->
                    <Grid Background="{Binding Converter={StaticResource HashCodeToRandomColorBrush}}"
               Height="120"
               Width="120"
               CornerRadius="5"
               Margin="5">
                        <!-- Some Text -->
                        <TextBlock Style="{StaticResource HeadlineLarge}"
                       Foreground="White"
                       HorizontalAlignment="Center"
                       VerticalAlignment="Center"
                       Text="{Binding }" />
                    </Grid>
                </DataTemplate>
            </GridView.ItemTemplate>
        </GridView>
```

Microsoft.UI.Xaml.Markup.XamlParseException
  HResult=0x802B000A
  Message=XAML parsing failed.
  Source=WinRT.Runtime
  StackTrace:
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr)
   at WinRT.ExceptionHelpers.ThrowExceptionForHR(Int32 hr)
   at ABI.Microsoft.UI.Xaml.IApplicationStaticsMethods.LoadComponent(IObjectReference _obj, Object component, Uri resourceLocator, ComponentResourceLocation componentResourceLocation)
   at Microsoft.UI.Xaml.Application.LoadComponent(Object component, Uri resourceLocator, ComponentResourceLocation componentResourceLocation)
   at App1UNO.MainPage.InitializeComponent() in D:\Private\Projects\csharp\_TEST_\App1UNO\App1UNO.Windows\obj\x86\Debug\net7.0-windows10.0.19041.0\MainPage.g.i.cs:line 37

### Expected behavior

well at least you could test it. Since there is many good compoents on uno gallery i would expect its working..

### How to reproduce it (as minimally and precisely as possible)

simple make a new project from VS 2022 IDE from the WinAppSDK new template  and drop sample of GridView in MainView.xaml and application wont run.

### Workaround

none

### IDE

Visual Studio 2022

### IDE version

17.4

### Relevant plugins

none

### Anything else we need to know?

Many of the components from unogallery does not run with net7. Many errors like xaml parsing errors

![image](https://user-images.githubusercontent.com/1000729/202909507-b7333750-0601-42af-b566-73ebb8feefee.png)

**Comment by jeromelaban** [comment]
Thanks for the report. 

The code you're seeing here depends on a few other resources that you do not have in your app, like [`HashCodeToRandomColorBrush`](https://github.com/unoplatform/Uno.Gallery/blob/7b37b225839b45d74e931f772c99760986d32f7a/Uno.Gallery/Uno.Gallery.UWP/Views/Converters.xaml#L32-L33) (defined in Uno.Gallery) or [`HeadlineLarge`](https://github.com/unoplatform/Uno.Themes/blob/c4ef74746a4f06924c2554daef08f9a0137bf666/src/library/Uno.Material/Styles/Controls/v2/TextBlock.xaml#L66-L75) (defined by Uno.Material), which makes the app fail to load as a result.

You can see the same code [here in the playground](https://playground.platform.uno/#b1ec36ff), though you will not see the colors because the playground does not have `HashCodeToRandomColorBrush` either. You can import this particular converter or remove it completely.

**Reply by michalss** [reply]
but i don't talking about styles im talking about that parsing error in general, where application crashes. it has nothing to do with specific styling..

**Reply by michalss** [reply]
Microsoft.UI.Xaml.Markup.XamlParseException
HResult=0x802B000A
Message=XAML parsing failed.
Source=WinRT.Runtime
StackTrace:
at WinRT.ExceptionHelpers.g__Throw|20_0(Int32 hr)
at WinRT.ExceptionHelpers.ThrowExceptionForHR(Int32 hr)
at ABI.Microsoft.UI.Xaml.IApplicationStaticsMethods.LoadComponent(IObjectReference obj, Object component, Uri resourceLocator, ComponentResourceLocation componentResourceLocation)
at Microsoft.UI.Xaml.Application.LoadComponent(Object component, Uri resourceLocator, ComponentResourceLocation componentResourceLocation)
at App1UNO.MainPage.InitializeComponent() in D:\Private\Projects\csharp_TEST\App1UNO\App1UNO.Windows\obj\x86\Debug\net7.0-windows10.0.19041.0\MainPage.g.i.cs:line 37

**Reply by michalss** [reply]
![image](https://user-images.githubusercontent.com/1000729/202984554-e94b4271-89ef-480f-8237-381a01ac6420.png)

**Comment by michalss** [comment]
This is error, is simple does not work. even if you remove all styles variables. Yes it does work in uno playground. But try to make a new project from VS template and add it in MainPage.xaml, it will crash..

**Comment by michalss** [comment]
here is proof. This is 100% bug

![image](https://user-images.githubusercontent.com/1000729/202984515-276ec39c-1168-4192-b6be-7450b193d43e.png)

**Reply by jeromelaban** [reply]
Thanks for the update. Can you provide a repro sample that we can test? Note that when running under WinAppSDK, Uno is not present.

**Reply by michalss** [reply]
Thank you, i will upload sample when i get home. However this is happening with all plaforms not just WinAppSDK.

**Reply by jeromelaban** [reply]
Thanks for the sample. It seems that WinUI does not support the syntax `ItemsSource="123456781234567812345678"`, where UWP did.

While this syntax is useful for samples, because it uses strings as an enumerable character source, it can be changed to this:

```xml
        <GridView HorizontalAlignment="Center">
            <x:String>1</x:String>
            <x:String>2</x:String>
            <x:String>3</x:String>
            <x:String>4</x:String>
```

Nonetheless, in general it's best to use an actual item source, [with an `x:Bind` or `Binding` source](https://learn.microsoft.com/en-us/windows/apps/design/controls/listview-and-gridview#add-items-to-a-listview-or-gridview-control).

**Comment by michalss** [comment]
[AAAUNOTest.zip](https://github.com/unoplatform/uno/files/10058128/AAAUNOTest.zip)
here it is
